Python/instrumentation.c

```diff
@@ -588,9 +589,10 @@ de_instrument(PyCodeObject *code, int i, int event)
         return;
     }
     CHECK(_PyOpcode_Deopt[deinstrumented] == deinstrumented);
-    *opcode_ptr = deinstrumented;
+    FT_ATOMIC_STORE_UINT8_RELAXED(*opcode_ptr, deinstrumented);
```

Comment:
Is `relaxed` strong enough? Do any consumers need to see the updates that are sequenced-before this store?

Reply:
In the deinstrumented case if we race and still see the instrumented opcode the data will still be there.  In the instrumented case I think we do need some more synchronization around the tooling metadata, if we see the instrumented opcode before seeing the writes to the data that'll be bad.

I had originally had synchronizeds reads/writes around that in https://github.com/python/cpython/commit/d8ea6bb937723ff7f5363ffa3bf58c81114028e1#diff-adaefb7da847260ef7aff6e66007bc83f5b226c5389a23e1c9ea622c3b02c419 but the thought was that the synchronization around `_PyFrame_GetCode(frame)->_co_instrumentation_version` would be sufficient (which is in the latest version of https://github.com/python/cpython/pull/116775).

But that doesn't actually seem to be good enough - there's still some rare TSAN failures around the instrumentation data.  But my plan is to push the synchronization there instead of on the opcode because I don't think we want to do acquire loads on the byte code which would be required to make a release write here meaningful.